accounts/abi/bind/base.go

```diff
@@ -424,6 +431,31 @@ func (c *BoundContract) createPrivateTransaction(tx *types.Transaction, payload
 	return privateTx
 }
 
+// (Quorum) createMarkerTx creates a new public privacy marker transaction for the given private tx, distributing tx to the specified privateFor recipients
+func (c *BoundContract) createMarkerTx(opts *TransactOpts, tx *types.Transaction, args PrivateTxArgs) (*types.Transaction, error) {
+	// Choose signer to sign transaction
+	if opts.Signer == nil {
+		return nil, errors.New("no signer to authorize the transaction with")
+	}
+	signedTx, err := opts.Signer(types.QuorumPrivateTxSigner{}, opts.From, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	hash, err := c.transactor.DistributeTransaction(ensureContext(opts.Context), signedTx, args)
+	if err != nil {
+		return nil, err
+	}
+
+	// Note: using isHomestead and isEIP2028 set to true, which may give a slightly higher gas value (but avoids making an API call to get the block number)
+	intrinsicGas, err := core.IntrinsicGas(common.Hex2Bytes(common.MaxPrivateIntrinsicDataHex), false, true, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return types.NewTransaction(signedTx.Nonce(), common.QuorumPrivacyPrecompileContractAddress(), tx.Value(), intrinsicGas, tx.GasPrice(), common.FromHex(hash)), nil
```

Comment:
@SatpalSandhu61 does the hash has exactly 32 bytes?

Reply:
It's 64 bytes.

Reply:
non-blocking: Besu uses 32 bytes I believe. as we are not doing the same implementation as Besu and we use a different pre-compile address, in the future we can handle a common approach. however, I ask if is useful to follow now the same size?

Reply:
Tessera returns a 64 byte hash, so we have to use that. If we wanted to use a 32 byte hash (either in just the PMT, or in general) then it would require major changes.

Reply:
ok, thanks. understood

Reply:
@SatpalSandhu61 We could have computed the intrinsic gas of the actual PMT payload `common.FromHex(hash)` so we do not need to make any assumption on the format of the enclave key. Was there a reason for taking the intrinsic gas of `MaxPrivateIntrinsicDataHex` instead? Thanks.

Reply:
@nmvalera Good point. It should really use the payload hash rather than `MaxPrivateIntrinsicDataHex`. I'll create a PR to amend that.